context/UserInfo.tsx

```diff
@@ -34,10 +33,10 @@ export function UserInfoProvider(props: UserInfoProviderProps) {
 
   const setUserCallback = (user: IUser) => setUser(user);
 
-  if (!user) return props.children; // return normal data
+  if (status === "loading" || (session && !user)) return <CircularLoader />;
```

Comment:
Loader will be shown when session is still being checked to see if authenticated/unauthenticated or if session is found but we have not loaded user info yet. user can very well be passed in as null, the protected/premium route logic handles that.